### PR TITLE
Correct corrections

### PR DESCRIPTION
**Pull request by prabhuramachandran**
Fix the kernel correction implementation for Bonet and Lok.  Also fix the taylor green problem so this works.

### CODE REVIEW

**Comment by prabhuramachandran** [comment]
The CRKSPH correction is completely wrong and will be fixed in a subsequent PR.

File: pysph/sph/wc/kernel_correction.py [hunk 1]
```diff
@@ -187,12 +194,58 @@ def loop_all(self, d_idx, d_x, d_y, d_z, d_h, s_x, s_y, s_z, s_h, SPH_KERNEL,
...
+
+class MixedGradientCorrection(Equation):
+    r"""**Kernel Gradient Correction**
+
+    .. math::
+            \nabla \tilde{W}_{ab} = L_{a}\nabla W_{ab}
+
+    .. math::
+            L_{a} = \left(\sum \frac{m_{b}}{\rho_{b}} \nabla W_{ab}
+            \mathbf{\times}x_{ba} \right)^{-1}
```
Comment (abhinavmuta) [inline pysph/sph/wc/kernel_correction.py]: change '\times' to `\otimes'

File: pysph/sph/wc/kernel_correction.py [hunk 2]
```diff
@@ -187,12 +194,58 @@ def loop_all(self, d_idx, d_x, d_y, d_z, d_h, s_x, s_y, s_z, s_h, SPH_KERNEL,
...
+    References
+    ----------
+    .. [Bonet and Lok, 1999] Bonet, J. and Lok T.-S.L. (1999)
```
Comment (abhinavmuta) [inline pysph/sph/wc/kernel_correction.py]: change "[Bonet and Lok, 1999]" to something which don't have any spaces

File: pysph/sph/wc/kernel_correction.py [hunk 3]
```diff
@@ -118,7 +122,7 @@ class MixedKernelCorrectionPreStep(Equation):
 
     .. math::
             L_{a} = \left(\sum_{b} V_{b}}\nabla \bar{W}_{ab}
```
Comment (abhinavmuta) [inline pysph/sph/wc/kernel_correction.py]: unwanted "}" after "V_{b}"

**Pull request merged**